dataflow/src/main/java/org/checkerframework/dataflow/cfg/AbstractCFGVisualizer.java

```diff
@@ -0,0 +1,337 @@
+package org.checkerframework.dataflow.cfg;
+
+import java.util.*;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.analysis.*;
+import org.checkerframework.dataflow.cfg.block.*;
+import org.checkerframework.dataflow.cfg.node.Node;
+
+/**
+ * An abstract class of control flow graph visualizer. To achieve this abstract class, you need to
+ * implement some of the methods in {@link CFGVisualizer}.
+ *
+ * <p>Examples: {@link DOTCFGVisualizer} and {@link StringCFGVisualizer}.
+ */
+public abstract class AbstractCFGVisualizer<
+                A extends AbstractValue<A>, S extends Store<S>, T extends TransferFunction<A, S>>
+        implements CFGVisualizer<A, S, T> {
+
+    protected boolean verbose;
+
+    @Override
+    public void init(Map<String, Object> args) {
+        {
```

Comment:
This local scope no longer is necessary - it's the only thing happening in this method.